executable_schema.go

```diff
@@ -79,32 +80,49 @@ func (s *ExecutableSchema) UpdateSchema(forceRebuild bool) error {
 		}
 	}()
 
-	for url, s := range s.Services {
-		logger := log.WithField("url", url)
-		updated, err := s.Update()
-		if err != nil {
-			promServiceUpdateErrorCounter.WithLabelValues(s.ServiceURL).Inc()
-			promServiceUpdateErrorGauge.WithLabelValues(s.ServiceURL).Set(1)
-			invalidSchema, forceRebuild = true, true
-			logger.WithError(err).Error("unable to update service")
-			// Ignore this service in this update
-			continue
-		}
-		promServiceUpdateErrorGauge.WithLabelValues(s.ServiceURL).Set(0)
-		logger = log.WithFields(log.Fields{
-			"version": s.Version,
-			"service": s.Name,
-		})
-
-		if updated {
-			logger.Info("service was updated")
-			updatedServices = append(updatedServices, s.Name)
-		}
+	// Only fetch at most 64 services in parallel
+	rateLimit := make(chan struct{}, 64)
```

Comment:
Would [`errgroup.SetLimit`](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.SetLimit) provide the same behaviour as the `rateLimit` channel?

Reply:
Technically it would be slightly different, as the call to `errgroup.Go` would block on the main thread, instead of the green thread itself blocking, but I don't think that makes a material difference, so I can use that if you prefer.

Reply:
I think it's clearer to read and it shouldn't matter which goroutine blocks, since the `errgroup.Wait` call blocks afterwards anyway

Reply:
Yea agreed; apparently this method was only added in September of 2022, hence why I wasn't aware of it 😅 Certainly a lot more convenient!